apps/web/app/(app)/environments/[environmentId]/settings/product/DeleteProduct/DeleteProduct.tsx

```diff
@@ -0,0 +1,38 @@
+import { getAvailableProducts } from "@formbricks/lib/services/product";
+import { getProfile } from "@formbricks/lib/services/profile";
+import { getTeamByEnvironmentId } from "@formbricks/lib/services/team";
+import { getMembersByTeamId } from "@formbricks/lib/services/membership";
+import { TProduct } from "@formbricks/types/v1/product";
+import DeleteProductRender from "@/app/(app)/environments/[environmentId]/settings/product/DeleteProduct/DeleteProductRender";
+import { getServerSession } from "next-auth";
+import { authOptions } from "@/app/api/auth/[...nextauth]/authOptions";
+
+type DeleteProductProps = {
+  environmentId: string;
+  product: TProduct;
+};
+
+export default async function DeleteProduct({ environmentId, product }: DeleteProductProps) {
+  const session = await getServerSession(authOptions);
+
+  const profile = session ? await getProfile(session.user.id) : null;
+  const team = await getTeamByEnvironmentId(environmentId);
```

Comment:
We do a lot of complicated database operations "just" to determine the role of the current user within the team. We also already have something we can use for that in the session:
```json
{
  "@formbricks/web:go": {
    "user": {
      "name": "Matti",
      "email": "mail@matti.sh",
      "id": "cll6p6pdo000019t83t36a3te",
      "createdAt": "2023-08-11T14:42:55.356Z",
      "onboardingCompleted": true,
      "teams": [
        {
          "id": "cll6p6pdo000119t84cao4qch",
          "role": "owner",
          "plan": "free"
        }
      ]
    }
  }
}
```

Reply:
this is really cool, wish I had inspected the session object more 😅